### ISSUE TITLE
Subsequent calls to mongoengine.connect does not change the current database connection

### ISSUE THREAD

**Issue by apiguy** [hmarr/mongoengine#73]
After an initial call to mongoengine.connect(), any subsequent call will have no effect because the mongoengine.connection._get_db() function only generates a new database when the global _db is None. No code currently exists to reset the value of the global _db.

_Example:_

```
>>> import mongoengine
>>> mongoengine.connect("db1")
Database(Connection('localhost', 27017), u'db1')
>>> mongoengine.connect("db2")
Database(Connection('localhost', 27017), u'db1')
```

_Simple Fix_

This issue has a very simple workaround, which is probably all that is currently necessary since I expect the upcoming multiple database support is going to change how this section of code works fairly dramatically.

in connection.py change this:

```
def connect(db, username=None, password=None, **kwargs):
    """Connect to the database specified by the 'db' argument. Connection 
    settings may be provided here as well if the database is not running on
    the default port on localhost. If authentication is needed, provide
    username and password arguments as well.
    """
    global _connection_settings, _db_name, _db_username, _db_password
    _connection_settings.update(kwargs)
    _db_name = db
    _db_username = username
    _db_password = password
    return _get_db()
```

to this:

```
def connect(db, username=None, password=None, **kwargs):
    """Connect to the database specified by the 'db' argument. Connection 
    settings may be provided here as well if the database is not running on
    the default port on localhost. If authentication is needed, provide
    username and password arguments as well.
    """
    global _connection_settings, _db_name, _db_username, _db_password , _db
    _connection_settings.update(kwargs)
    _db_name = db
    _db_username = username
    _db_password = password
    _db = None # Force it to change databases
    return _get_db()
```

**Comment by almighty-loaf** [comment]
Wow, thanks for the awesome answer! To extend this issue to my problem (using the connect('my collection') to see if the mongod server is running), you can remove the
        if _connection is None:
lines in both __get_db_ and __get_connection_ to force it to either successfully reconnect every time or throw an exception.

**Comment by hmarr** [comment]
Thanks for the solid problem description - brought in a commit (with some changes) that should fix these issues (92471445ec5aede3d6ffb14e6b63026156c6d623). Would you mind checking that this solves the problems? This is on the v0.4 branch.

Cheers,
Harry
